app/controllers/comments_controller.rb

```diff
@@ -9,7 +9,35 @@ def index
 
   api :POST, '/comments'
   def create
-    comment = Comment.create(comment_params)
+    body = comment_params[:body]
+    comment = Comment.create(body: body, user_id: comment_params[:user_id], bucket_id: comment_params[:bucket_id])
+
+    # convert markdown body to html
+    markdown = Redcarpet::Markdown.new(Redcarpet::Render::HTML, extensions = {})
+    red_body = markdown.render(body)
+
+    # break the html into fragments
+    noko_body = Nokogiri::HTML.fragment(red_body)
+
+    # get an array of all the user ids from the link values
+    user_links = noko_body.css("a").map {|a|
+      a.attributes.values[0].value.sub! 'uid:', ''
+    }
+
+    # format the links with the correct base url
+    noko_body.css("a").map do |a|
+      a['href'].sub! 'uid:', ''
+      a.attributes["href"].value = "#{Rails.application.config.action_mailer.default_url_options[:host]}/##{a['href'].sub! 'uid:', '/users/'}"
+    end
+
+    if user_links
+      email_body = noko_body.to_html
+      bucket = Bucket.find(comment_params[:bucket_id])
+      user_links.each do |userId|
+        user = User.find(userId)
+        UserMailer.notify_member_that_they_were_mentioned(author: current_user, member: user, bucket: bucket, body: email_body).deliver_later
+      end
+    end
```

Comment:
I wouldn't put the code that converts from markdown to HTML in the controller, but rather in the `user_mailer` code that generates the mail.

Reply:
@chime-mu It's not always the case that the `user_mailer` would get called though. If no one is mentioned in the comment then no mails would get sent. Because of this, I'm not sure that it makes sense to me to put this code in the `user_mailer`. Does that make sense? Or is there something I'm not seeing?

Reply:
Why would you convert to HTML if no mail should be sent?